ide/coqide/shared_WIN32.c.in

```diff
@@ -0,0 +1,16 @@
+#define _WIN32_WINNT 0x0501  /* Cf below, we restrict to  */
+
+#include <caml/mlvalues.h>

```

Comment:
We already have a C stub file per OS, is there a solid reason to add a new one? I think it's not very good practice from the OCaml world to have a lot of C stubs lying around (especially given that they're preprocessed).

Reply:
I didn't want to add the second one.  `win32_cvtpid` is used both in CoqIDE and in coqidetop.  In the dune file, `coqide_WIN32.ml.in` is part of library `coqide_gui`.  When I included `win32_cvtpid` in `coqide_WIN32.c.in` as @rgrinberg [suggested](https://github.com/coq/coq/pull/16139#issuecomment-1146852648), I got very strange behavior in CoqIDE (delay of a second or so for each character entered).  I'm guessing that all the files in the library were not merely linked, but in fact loaded.  So their `let () = ...` expressions executed rather than just the ones needed, e.g. as in `coqide_main.ml`, which is bad.

If there's a way to link specific modules without loading everything in the library, I'm all for it.  I didn't understand Rudi's suggestion [here](https://github.com/coq/coq/pull/16139#discussion_r889754180) well enough to try it.

Reply:
I figured out a way to avoid the need for `shared_WIN32.c.in` and its friends.  Now there's `os_specific.ml` which has the references to C code (usable in both coqide and coqidetop) and `gui_specific.ml`, containing items that depend on GTK and the GUI.  Tedious to sort these out.  If you're OK with this, I'll squash before the merge.

Reply:
I spoke too soon.  It appears that `Glib.Io.channel_of_descr_socket` in GTK is Windows-only.  Whatever library contains that can't be linked into `coqidetop`, while `win32_cvtpid` must be linked into both `coqide` and `coqidetop`.  It appears they need to be two separate libraries.  Let's go with the first commit.  I'll remove the second one momentarily.